qiskit/quantum_info/operators/symplectic/base_pauli.py

```diff
@@ -0,0 +1,639 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Optimized list of Pauli operators
+"""
+# pylint: disable=invalid-name, abstract-method
+
+import copy
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.barrier import Barrier
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+
+class BasePauli(BaseOperator):
+    r"""Symplectic representation of a list of N-qubit Paulis.
+
+    Base class for Pauli and PauliList.
```

Comment:
Are you planning to implement symplectic PauliList in another PR?

Reply:
Yes, this PR is already quite large!